src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -193,6 +205,42 @@ internal static (HttpMessageHandler Pipeline, FeatureFlag FeatureFlags) CreatePi
             return (Pipeline: httpPipeline, FeatureFlags: handlerFlags);
         }
 
+        /// <summary>
+        /// Gets a platform's native http handler i.e. NSUrlSessionHandler for Xamarin.iOS, AndroidClientHandler for Xamarin.Android and HttpClientHandler for others.
+        /// </summary>
+        /// <param name="proxy">The proxy to be used with created client.</param>
+        /// <returns>
+        /// 1. NSUrlSessionHandler for Xamarin.iOS 
+        /// 2. AndroidClientHandler for Xamarin.Android.
+        /// 3. HttpClientHandler for other platforms.
+        /// </returns>
+        internal static HttpMessageHandler GetNativePlatformHttpHandler(IWebProxy proxy = null)
+        {
+#if iOS
+            return new NSUrlSessionHandler { AllowAutoRedirect = false };
+#elif ANDROID
+            return new Xamarin.Android.Net.AndroidClientHandler { Proxy = proxy, AllowAutoRedirect = false, AutomaticDecompression = DecompressionMethods.None };
+#else
+            return new HttpClientHandler { Proxy = proxy, AllowAutoRedirect = false, AutomaticDecompression = DecompressionMethods.None };
+#endif
+        }
+
+        /// <summary>
+        /// Removes the specified targetHandlerType from the provided list of delegatingHandlers.
+        /// </summary>
+        /// <param name="delegatingHandlers">The list of <see cref="DelegatingHandler"/> to remove a <see cref="DelegatingHandler"/> type from.</param>
+        /// <param name="targetHandlerType">The <see cref="DelegatingHandler"/> type to remove.</param>
+        /// <returns></returns>
+        internal static bool RemoveHandler(IList<DelegatingHandler> delegatingHandlers, Type targetHandlerType)
+        {
+            if (delegatingHandlers == null)
+            {
+                throw new ArgumentNullException(nameof(delegatingHandlers), "DelegatingHandler list contains null item.");
+            }
+
+            return delegatingHandlers.Remove(delegatingHandlers.FirstOrDefault((h) => h.GetType().Equals(targetHandlerType)));
```

Comment:
We prevent adding of duplicate handlers by types in [L195](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/81129d8f122e94b4d2ae73f9a4dffdeaf5b11c89/src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs#L195).

It shouldn't break anything since it's internal and is only used when targeting iOS to remove our inbuilt CompressionHandler from the delegating handlers list.